packages/flutter/lib/src/widgets/image_filter.dart

```diff
@@ -2,76 +2,119 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:ui';
+import 'dart:ui' as ui;
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/rendering.dart';
 
 import 'framework.dart';
 
+// Examples can assume:
+// late AnimationController controller;
+
 /// Applies an [ImageFilter] to its child.
 ///
+/// For example, an [ImageFilter.blur] can be applied to blur a child that
+/// needs to be obscured, or an [ImageFilter.matrix] can be used to apply a
+/// bitmap transform to a child at a small trade-off of quality for performance.
+///
+/// {@tool snippet}
+///
+/// This example makes the text blurry:
+///
+/// ```dart
+/// ImageFiltered(
+///   imageFilter: ui.ImageFilter.blur(sigmaX: 5.0, sigmaY: 5.0),
+///   child: const Text('I am blurry'),
+/// )
+/// ```
+/// {@end-tool}
+///
+/// {@tool snippet}
+///
+/// This example makes the text (or other complicated widget) rotate
+/// around its center based on an [AnimationController] value.
+///
+/// ```dart
+/// ImageFiltered(
+///   imageFilterCallback: (Rect bounds) => ui.ImageFilter.matrix(
```

Comment:
(as also discussed in the design doc): A mechanism is needed to inform the ImageFiltered that the imageFilterCallback needs to be evaluated again because `controller.value` changed.